### DISCUSSION TITLE
Email says `{user} pushed 0 commits.` for force-push that drops commits

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#57559] [Pull Requests]
### Select Topic Area

Bug

### Body

We have a number of repositories where the last commit is a test commit, and the last thing someone does is a force push to drop the commit.

That results in an email with:

```
@someone pushed 0 commits.
—
View it on GitHub or unsubscribe.
You are receiving this because you are subscribed to this thread.
```

This message is unhelpful, and clicking clicking `View it on GitHub` yields:

```
[Title: Commit range not found · Pull Request #{PR-number} · {Org/Repo}]

We went looking everywhere, but couldn’t find those commits.
Sometimes commits can disappear after a force-push. Head back to the latest changes here.
```

Which is *also* unhelpful.

It would be more helpful and more accurate to say:

```
@someone pushed -1 commits.
—
View it on GitHub or unsubscribe.
You are receiving this because you are subscribed to this thread.
```

or

```
@someone pushed -1 commit.
—
View it on GitHub or unsubscribe.
You are receiving this because you are subscribed to this thread.
```

or

```
@someone dropped 1 commits.
—
View it on GitHub or unsubscribe.
You are receiving this because you are subscribed to this thread.
```

or

```
@someone dropped 1 commit.
—
View it on GitHub or unsubscribe.
You are receiving this because you are subscribed to this thread.
```

And have the link properly show a thing with a list of commits and show the dropped commits in github diff pink.

Note that this is not the same as #29969.
I can't tell if it's the same as #7230 (there isn't enough analysis, and a quick look at the referenced PR didn't give me any obvious insight).

**Comment by setharnold** [comment]
Or:

`@someone dropped commit abcdef1`

`@someone dropped commits abcdef1 bcdef12 cdef123`
